### ISSUE TITLE
command 'HookyQR.minify' not found

### ISSUE THREAD

**Issue by nkotsioris** [HookyQR/VSCodeMinify#2]
I have an issue, it appears in both the computers I work on:

Please find the exception information below:
`[Extension Host] Here is the error stack:  Error: Cannot find module 'amdefine'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Function.O.n._load (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:11:30541)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (C:\Users\n.kotsioris\.vscode\extensions\HookyQR.minify-0.2.1\node_modules\source-map\lib\source-map\source-map-generator.js:8:18)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)`

**Comment by gbellini79** [comment]
Same problem here after upgrading to Visual Studio Code 1.4

**Comment by ozmartian** [comment]
not working for me either... same result as already posted :-(

**Comment by dorianmuthig** [comment]
The issue isn't with the VScode 1.4 update, but with the 0.2.1 update of the extension, which updated dependent components but not the components the new versions now depend on. To fix it close VScode, open a command line terminal and go into the extension folder `HookyQR.minify-0.2.1` under `%USERPROFILE%\.vscode\extensions` on Windows or `~/.vscode/extensions` on macOS/Linux and run `npm install`. It should now work. This is assuming you have the node package manager installed on your system, if you don't you may have to install it first.

**Comment by nkotsioris** [comment]
@dorianmuthig Thanks, that solved the problem, although I still consider it an issue with the package.

**Comment by justinlabenne** [comment]
@dorianmuthig, I followed the steps you have provided and this did not work for me. I have Node installed on Windows (I have tried the x64 and x86 versions) but I continue to receive the "HookyQR.minify" not found error. Here is the output below when running "npm install" against the "HookyQR.minify-0.2.1" extension folder:

```
C:\Users\xxx\.vscode\extensions\HookyQR.minify-0.2.1>npm install

npm WARN deprecated vscode@0.10.6: Please update to version 0.11.x for improved typings and test support
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated to-iso-string@0.0.2: to-iso-string has been deprecated, use@segment/to-iso-string instead.
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
vscode@0.10.6 node_modules\vscode
├── glob@5.0.15 (path-is-absolute@1.0.0, inherits@2.0.1, once@1.3.3, inflight@1.0.5, minimatch@3.0.3)
├── source-map-support@0.3.3 (source-map@0.1.32)
└── mocha@2.5.3 (escape-string-regexp@1.0.2, diff@1.4.0, commander@2.3.0, growl@
1.9.2, supports-color@1.2.0, to-iso-string@0.0.2, debug@2.2.0, mkdirp@0.5.1, glob@3.2.11, jade@0.26.3)
```
Any thoughts you can provide as to what I am missing are much appreciated.

**Comment by HookyQR** [comment]
There are a bunch of problems with how the VS Code packager is handling dependancies. Don't upgrade to 0.2.2 yet. It doesn't work either. I'm working on a fix.

In the meantime, use `npm install --production` instead of just `npm install` to avoid the vscode dependancy getting added.

Apologies for the current situation.

**Comment by ozmartian** [comment]
@HookyQR 
Just installed the recently released 0.2.2 and its fixed the issue for me.. all working again on my Arch Linux box.

thanks for that!

**Comment by HookyQR** [comment]
OMG @ozmartian it didn't work for me!! Glad to hear it.

Wait. It did work. I was testing it on a file unsupported features.

So, 0.2.2 solves the problem. Again, sorry for the stuff around. And thanks to @dorianmuthig for providing directions for a fix in the meantime.
